[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/hien078/applied-mathematics-foundation/blob/master/calculus/14_vector_calculus_field_theorems/first_principles.ipynb)

# Topic 14: Vector Calculus & Field Theorems — First-Principles Theory & Derivations

**Author:** Advanced Calculus & Mathematical Modeling Team  
**Module Type:** Comprehensive Core Curriculum  

---

## 1. First-Principles Intuition & Motivation

In single-variable calculus, we study functions $f: \mathbb{R} \to \mathbb{R}$ where instantaneous rates of change are given by $f'(x)$ and total accumulation along an interval $[a, b]$ is given by the Fundamental Theorem of Calculus:

$$
\int_a^b f'(x)\,\mathrm{d}x = f(b) - f(a)
$$

In multivariable scalar calculus, partial derivatives $\frac{\partial f}{\partial x_i}$ and gradients $\nabla f$ measure local sensitivity across multiple space directions. However, physical reality—from fluid motion and electromagnetic forces to gravitational fields and optimization trajectory velocities—is governed by spatially distributed multi-component velocity or force vectors: **vector fields**.

Vector calculus provides the rigorous framework to answer two fundamental structural questions:
1. **Local Microscopic Behavior**: At an infinitesimal point in space, is the field expanding/contracting (creating or destroying density)? Is it spinning/rotating (creating micro-vorticity)?
2. **Global Macroscopic Accumulation**: How does local expansion or rotation over an entire region accumulate along its boundary?

The grand synthesis of vector calculus demonstrates that all continuous field operations (gradient, curl, divergence) and all fundamental integration theorems (Fundamental Theorem of Line Integrals, Green's Theorem, Stokes' Theorem, Gauss's Divergence Theorem) are expressions of a single topological principle: **the integral of a derivative over a domain equals the integral of the original quantity over the boundary of that domain**.

---

## 2. Rigorous Mathematical Definitions & Theorem Statements

### 2.1 Vector Fields
Let $U \subseteq \mathbb{R}^n$ be an open set. A **vector field** on $U$ is a mapping $\mathbf{F}: U \to \mathbb{R}^n$ that assigns to every point $\mathbf{x} \in U$ a vector $\mathbf{F}(\mathbf{x}) \in \mathbb{R}^n$.

In $\mathbb{R}^3$, we represent $\mathbf{F}(\mathbf{x}) = \mathbf{F}(x, y, z)$ as:

$$
\mathbf{F}(x, y, z) = P(x, y, z)\mathbf{i} + Q(x, y, z)\mathbf{j} + R(x, y, z)\mathbf{k} = \begin{pmatrix} P(x, y, z) \\ Q(x, y, z) \\ R(x, y, z) \end{pmatrix}
$$

where $P, Q, R: U \to \mathbb{R}$ are scalar component functions. A vector field is of class $C^k$ if its component functions are $k$-times continuously differentiable.

---

### 2.2 Line Integrals

#### Scalar Line Integral
Let $C \subset \mathbb{R}^n$ be a smooth curve parameterized by $\mathbf{r}(t)$, for $t \in [a, b]$, with $\mathbf{r}'(t) \neq \mathbf{0}$. The **line integral of a scalar function** $f: \mathbb{R}^n \to \mathbb{R}$ along $C$ is defined as:

$$
\int_C f\,\mathrm{d}s = \int_a^b f(\mathbf{r}(t)) \|\mathbf{r}'(t)\|\,\mathrm{d}t
$$

where $\mathrm{d}s = \|\mathbf{r}'(t)\|\,\mathrm{d}t$ is the differential arc length element.

#### Vector Line Integral (Work / Circulation)
The **line integral of a vector field** $\mathbf{F}$ along an oriented smooth curve $C$ parameterized by $\mathbf{r}(t)$, $t \in [a, b]$, is defined as:

$$
\int_C \mathbf{F} \cdot \mathrm{d}\mathbf{r} = \int_a^b \mathbf{F}(\mathbf{r}(t)) \cdot \mathbf{r}'(t)\,\mathrm{d}t = \int_C \mathbf{F} \cdot \mathbf{T}\,\mathrm{d}s
$$

where $\mathbf{T}(t) = \frac{\mathbf{r}'(t)}{\|\mathbf{r}'(t)\|}$ is the unit tangent vector to $C$. If $C$ is a closed curve ($\mathbf{r}(a) = \mathbf{r}(b)$), the integral is written as $\oint_C \mathbf{F} \cdot \mathrm{d}\mathbf{r}$ and measures the **circulation** of $\mathbf{F}$ around $C$.

---

### 2.3 Conservative Vector Fields & Potential Functions

#### Definition

> A vector field $\mathbf{F}: U \to \mathbb{R}^n$ is called **conservative** (or path-independent) on $U$ if there exists a differentiable scalar field $f: U \to \mathbb{R}$ (called the **scalar potential**) such that:
>
> $$
> \mathbf{F} = \nabla f
> $$
>
> (In physics, $\mathbf{F} = -\nabla V$, where $V$ is potential energy).

#### Fundamental Theorem of Line Integrals (Gradient Theorem)

> Let $U \subseteq \mathbb{R}^n$ be open, $f: U \to \mathbb{R}$ be a $C^1$ scalar function, and $C$ be a piecewise smooth curve in $U$ parameterized from point $A = \mathbf{r}(a)$ to point $B = \mathbf{r}(b)$. Then:
>
> $$
> \int_C \nabla f \cdot \mathrm{d}\mathbf{r} = f(B) - f(A)
> $$

#### Path Independence Equivalence Theorem

> Let $\mathbf{F}: U \to \mathbb{R}^n$ be a continuous vector field on an open connected domain $U$. The following statements are logically equivalent:
> 1. $\mathbf{F}$ is conservative (i.e., $\mathbf{F} = \nabla f$ for some $C^1$ potential $f$).
> 2. For any two points $A, B \in U$, $\int_{C_1} \mathbf{F} \cdot \mathrm{d}\mathbf{r} = \int_{C_2} \mathbf{F} \cdot \mathrm{d}\mathbf{r}$ for any two piecewise smooth paths $C_1, C_2$ in $U$ from $A$ to $B$.
> 3. For every closed curve $C$ in $U$, $\oint_C \mathbf{F} \cdot \mathrm{d}\mathbf{r} = 0$.

#### Cross-Partial Symmetry & Simple Connectivity
If $\mathbf{F} = P\mathbf{i} + Q\mathbf{j} + R\mathbf{k}$ is $C^1$ and conservative on $U$, then by Clairaut's theorem:

$$
\frac{\partial P}{\partial y} = \frac{\partial Q}{\partial x}, \quad \frac{\partial P}{\partial z} = \frac{\partial R}{\partial x}, \quad \frac{\partial Q}{\partial z} = \frac{\partial R}{\partial y}
$$

Conversely, if $U \subseteq \mathbb{R}^3$ is **simply connected** (every loop can be continuously contracted to a point within $U$) and $\nabla \times \mathbf{F} = \mathbf{0}$ everywhere in $U$, then $\mathbf{F}$ is conservative on $U$.

---

### 2.4 Green's Theorem in the Plane

Let $D \subset \mathbb{R}^2$ be a bounded domain whose boundary $C = \partial D$ consists of a finite number of simple, closed, piecewise smooth curves oriented positively (counterclockwise, so $D$ stays on the left). Let $P, Q: \bar{D} \to \mathbb{R}$ be $C^1$ functions.

#### Tangential (Circulation-Curl) Form

$$
\oint_C (P\,\mathrm{d}x + Q\,\mathrm{d}y) = \iint_D \left( \frac{\partial Q}{\partial x} - \frac{\partial P}{\partial y} \right) \mathrm{d}A
$$

#### Normal (Flux-Divergence) Form
Let $\mathbf{F} = P\mathbf{i} + Q\mathbf{j}$ and let $\mathbf{n}$ be the outward unit normal to $C$. Then:

$$
\oint_C \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}s = \oint_C (-Q\,\mathrm{d}x + P\,\mathrm{d}y) = \iint_D \left( \frac{\partial P}{\partial x} + \frac{\partial Q}{\partial y} \right) \mathrm{d}A = \iint_D (\nabla \cdot \mathbf{F})\,\mathrm{d}A
$$

---

### 2.5 Surface Integrals & Vector Flux

Let $S \subset \mathbb{R}^3$ be a smooth parametric surface defined by $\mathbf{r}(u, v) = x(u, v)\mathbf{i} + y(u, v)\mathbf{j} + z(u, v)\mathbf{k}$ over a parameter domain $D \subset \mathbb{R}^2$. The normal vector element is given by:

$$
\mathbf{N}(u, v) = \frac{\partial \mathbf{r}}{\partial u} \times \frac{\partial \mathbf{r}}{\partial v}
$$

#### Scalar Surface Integral
The integral of a scalar function $f(x, y, z)$ over $S$ is:

$$
\iint_S f\,\mathrm{d}S = \iint_D f(\mathbf{r}(u, v)) \|\mathbf{N}(u, v)\|\,\mathrm{d}u\,\mathrm{d}v
$$

where $\mathrm{d}S = \|\mathbf{N}(u, v)\|\,\mathrm{d}u\,\mathrm{d}v$ is the differential surface area element.

#### Vector Surface Integral (Flux)
Let $\mathbf{n} = \frac{\mathbf{N}(u, v)}{\|\mathbf{N}(u, v)\|}$ be the unit normal vector defining the orientation of $S$. The **flux** of a vector field $\mathbf{F}$ through $S$ is:

$$
\iint_S \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}S = \iint_D \mathbf{F}(\mathbf{r}(u, v)) \cdot \left( \frac{\partial \mathbf{r}}{\partial u} \times \frac{\partial \mathbf{r}}{\partial v} \right) \mathrm{d}u\,\mathrm{d}v
$$

---

### 2.6 Differential Operators: Gradient, Divergence, Curl

Let $\nabla = \mathbf{i}\frac{\partial}{\partial x} + \mathbf{j}\frac{\partial}{\partial y} + \mathbf{k}\frac{\partial}{\partial z}$ be the vector differential operator.

#### 1. Gradient ($\nabla f$)
For a scalar field $f: \mathbb{R}^3 \to \mathbb{R}$:

$$
\nabla f = \frac{\partial f}{\partial x}\mathbf{i} + \frac{\partial f}{\partial y}\mathbf{j} + \frac{\partial f}{\partial z}\mathbf{k}
$$

*Geometric Meaning*: Points in the direction of maximum rate of increase of $f$; magnitude $\|\nabla f\|$ is the maximum directional derivative.

#### 2. Divergence ($\nabla \cdot \mathbf{F}$)
For a vector field $\mathbf{F} = P\mathbf{i} + Q\mathbf{j} + R\mathbf{k}$:

$$
\nabla \cdot \mathbf{F} = \frac{\partial P}{\partial x} + \frac{\partial Q}{\partial y} + \frac{\partial R}{\partial z}
$$

*Coordinate-Free Definition (Limit of Flux Density)*:

$$
(\nabla \cdot \mathbf{F})(\mathbf{x}_0) = \lim_{\operatorname{Vol}(V) \to 0} \frac{1}{\operatorname{Vol}(V)} \oiint_{\partial V} \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}S
$$

*Physical Meaning*: Local volume expansion rate per unit volume. $\nabla \cdot \mathbf{F} \gt 0$ represents a source; $\nabla \cdot \mathbf{F} \lt 0$ represents a sink.

#### 3. Curl ($\nabla \times \mathbf{F}$)
For a vector field $\mathbf{F} = P\mathbf{i} + Q\mathbf{j} + R\mathbf{k}$:

$$
\nabla \times \mathbf{F} = \begin{vmatrix} \mathbf{i} & \mathbf{j} & \mathbf{k} \\ \frac{\partial}{\partial x} & \frac{\partial}{\partial y} & \frac{\partial}{\partial z} \\ P & Q & R \end{vmatrix} = \left( \frac{\partial R}{\partial y} - \frac{\partial Q}{\partial z} \right)\mathbf{i} + \left( \frac{\partial P}{\partial z} - \frac{\partial R}{\partial x} \right)\mathbf{j} + \left( \frac{\partial Q}{\partial x} - \frac{\partial P}{\partial y} \right)\mathbf{k}
$$

*Coordinate-Free Definition (Limit of Circulation Density)*:

$$
[(\nabla \times \mathbf{F})(\mathbf{x}_0)] \cdot \mathbf{n} = \lim_{\text{Area}(S) \to 0} \frac{1}{\text{Area}(S)} \oint_{\partial S} \mathbf{F} \cdot \mathrm{d}\mathbf{r}
$$

*Physical Meaning*: Axis and magnitude of local angular velocity / micro-vorticity of the field.

---

### 2.7 Stokes' Theorem

> Let $S \subset \mathbb{R}^3$ be an oriented, piecewise smooth surface bounded by a simple, closed, piecewise smooth curve $C = \partial S$ whose orientation is compatible with $S$ via the right-hand rule. Let $\mathbf{F}$ be a $C^1$ vector field on an open region containing $S$. Then:
>
> $$
> \oint_{\partial S} \mathbf{F} \cdot \mathrm{d}\mathbf{r} = \iint_S (\nabla \times \mathbf{F}) \cdot \mathbf{n}\,\mathrm{d}S
> $$

---

### 2.8 Gauss's Divergence Theorem

> Let $V \subset \mathbb{R}^3$ be a compact, solid region whose boundary $S = \partial V$ is a closed, piecewise smooth surface oriented with an outward unit normal vector field $\mathbf{n}$. Let $\mathbf{F}$ be a $C^1$ vector field defined on an open set containing $V$. Then:
>
> $$
> \oiint_{\partial V} \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}S = \iiint_V (\nabla \cdot \mathbf{F})\,\mathrm{d}V
> $$

---

### 2.9 Unified Differential Forms View (Exterior Calculus) — Preview

This section is a **preview, not a self-contained treatment**: the wedge product $\wedge$, the exterior derivative $\mathrm{d}$ on general $k$-forms, pullbacks, and orientation of a manifold with boundary are used below without being defined. For the full machinery and proof of the Generalized Stokes' Theorem, see Spivak, *Calculus on Manifolds*, Ch. 4–5.

In the language of differential forms (Spivak, *Calculus on Manifolds*):
- A scalar function is a 0-form $f$. Its exterior derivative is the 1-form $\mathrm{d}f = \frac{\partial f}{\partial x}\mathrm{d}x + \frac{\partial f}{\partial y}\mathrm{d}y + \frac{\partial f}{\partial z}\mathrm{d}z$ (Gradient).
- A vector field $\mathbf{F} = (P, Q, R)$ corresponds to a work 1-form $\omega^1_{\mathbf{F}} = P\mathrm{d}x + Q\mathrm{d}y + R\mathrm{d}z$. Its exterior derivative is the flux 2-form $\mathrm{d}\omega^1_{\mathbf{F}} = (\nabla \times \mathbf{F})_x \mathrm{d}y \wedge \mathrm{d}z + (\nabla \times \mathbf{F})_y \mathrm{d}z \wedge \mathrm{d}x + (\nabla \times \mathbf{F})_z \mathrm{d}x \wedge \mathrm{d}y$ (Curl).
- The vector field $\mathbf{F}$ also corresponds to a flux 2-form $\omega^2_{\mathbf{F}} = P\mathrm{d}y \wedge \mathrm{d}z + Q\mathrm{d}z \wedge \mathrm{d}x + R\mathrm{d}x \wedge \mathrm{d}y$. Its exterior derivative is the volume 3-form $\mathrm{d}\omega^2_{\mathbf{F}} = (\nabla \cdot \mathbf{F})\mathrm{d}x \wedge \mathrm{d}y \wedge \mathrm{d}z$ (Divergence).

All fundamental field theorems collapse into the single, elegant **Generalized Stokes' Theorem**:

$$
\int_{\partial \Omega} \omega = \int_{\Omega} \mathrm{d}\omega
$$

where $\Omega$ is a $k$-dimensional oriented manifold with boundary $\partial \Omega$, and $\omega$ is a $(k-1)$-form.

---

## 3. Step-by-Step Mathematical Proofs & Derivations

### Proof 1: The Fundamental Theorem of Line Integrals

> **Theorem**: If $\mathbf{F} = \nabla f$ on a smooth curve $C$ parameterized by $\mathbf{r}(t)$ for $t \in [a, b]$, then $\int_C \mathbf{F} \cdot \mathrm{d}\mathbf{r} = f(\mathbf{r}(b)) - f(\mathbf{r}(a))$.

*Proof*:
By definition of the line integral:

$$
\int_C \mathbf{F} \cdot \mathrm{d}\mathbf{r} = \int_a^b \mathbf{F}(\mathbf{r}(t)) \cdot \mathbf{r}'(t)\,\mathrm{d}t = \int_a^b \nabla f(\mathbf{r}(t)) \cdot \mathbf{r}'(t)\,\mathrm{d}t
$$

By the multivariable chain rule, the integrand is the total derivative of $f(\mathbf{r}(t))$ with respect to $t$:

$$
\frac{\mathrm{d}}{\mathrm{d}t} [f(\mathbf{r}(t))] = \nabla f(\mathbf{r}(t)) \cdot \mathbf{r}'(t)
$$

Substituting this into the integral yields:

$$
\int_C \nabla f \cdot \mathrm{d}\mathbf{r} = \int_a^b \frac{\mathrm{d}}{\mathrm{d}t} [f(\mathbf{r}(t))]\,\mathrm{d}t
$$

By the standard Single-Variable Fundamental Theorem of Calculus:

$$
\int_a^b \frac{\mathrm{d}}{\mathrm{d}t} [f(\mathbf{r}(t))]\,\mathrm{d}t = f(\mathbf{r}(b)) - f(\mathbf{r}(a))
$$

$$
\boxed{\int_C \nabla f \cdot \mathrm{d}\mathbf{r} = f(\mathbf{r}(b)) - f(\mathbf{r}(a))} \quad \blacksquare
$$

---

### Proof 2: Green's Theorem for a Type I and Type II Region
Let $D \subset \mathbb{R}^2$ be a region that is simultaneously Type I ($D = \{(x, y): a \le x \le b, g_1(x) \le y \le g_2(x)\}$) and Type II ($D = \{(x, y): c \le y \le d, h_1(y) \le x \le h_2(y)\}$). We wish to prove:

$$
\oint_{\partial D} (P\,\mathrm{d}x + Q\,\mathrm{d}y) = \iint_D \left( \frac{\partial Q}{\partial x} - \frac{\partial P}{\partial y} \right) \mathrm{d}A
$$

We establish this by proving two independent identities:
1. $\oint_{\partial D} P\,\mathrm{d}x = -\iint_D \frac{\partial P}{\partial y}\,\mathrm{d}A$
2. $\oint_{\partial D} Q\,\mathrm{d}y = \iint_D \frac{\partial Q}{\partial x}\,\mathrm{d}A$

#### Part 1: Proving $\oint_{\partial D} P\,\mathrm{d}x = -\iint_D \frac{\partial P}{\partial y}\,\mathrm{d}A$
Using Fubini's theorem on the double integral over the Type I representation of $D$:

$$
\iint_D \frac{\partial P}{\partial y}\,\mathrm{d}A = \int_a^b \left( \int_{g_1(x)}^{g_2(x)} \frac{\partial P}{\partial y}\,\mathrm{d}y \right) \mathrm{d}x
$$

Evaluating the inner integral via the Fundamental Theorem of Calculus:

$$
\int_{g_1(x)}^{g_2(x)} \frac{\partial P}{\partial y}\,\mathrm{d}y = P(x, g_2(x)) - P(x, g_1(x))
$$

Thus:

$$
\iint_D \frac{\partial P}{\partial y}\,\mathrm{d}A = \int_a^b P(x, g_2(x))\,\mathrm{d}x - \int_a^b P(x, g_1(x))\,\mathrm{d}x
$$

Now parameterize the boundary $\partial D = C_1 \cup C_2 \cup C_3 \cup C_4$:
- $C_1$: $y = g_1(x)$, $x$ goes from $a$ to $b$. Then $\int_{C_1} P\,\mathrm{d}x = \int_a^b P(x, g_1(x))\,\mathrm{d}x$.
- $C_2$: Vertical line segment $x = b$, $y$ goes from $g_1(b)$ to $g_2(b)$. Since $\mathrm{d}x = 0$, $\int_{C_2} P\,\mathrm{d}x = 0$.
- $C_3$: $y = g_2(x)$, $x$ goes from $b$ to $a$. Thus $\int_{C_3} P\,\mathrm{d}x = \int_b^a P(x, g_2(x))\,\mathrm{d}x = -\int_a^b P(x, g_2(x))\,\mathrm{d}x$.
- $C_4$: Vertical line segment $x = a$, $y$ goes from $g_2(a)$ to $g_1(a)$. Since $\mathrm{d}x = 0$, $\int_{C_4} P\,\mathrm{d}x = 0$.

Summing the line integrals around $C = \partial D$:

$$
\oint_C P\,\mathrm{d}x = \int_a^b P(x, g_1(x))\,\mathrm{d}x - \int_a^b P(x, g_2(x))\,\mathrm{d}x = -\iint_D \frac{\partial P}{\partial y}\,\mathrm{d}A
$$

#### Part 2: Proving $\oint_{\partial D} Q\,\mathrm{d}y = \iint_D \frac{\partial Q}{\partial x}\,\mathrm{d}A$
Using Fubini's theorem on the Type II representation of $D$:

$$
\iint_D \frac{\partial Q}{\partial x}\,\mathrm{d}A = \int_c^d \left( \int_{h_1(y)}^{h_2(y)} \frac{\partial Q}{\partial x}\,\mathrm{d}x \right) \mathrm{d}y = \int_c^d \Big( Q(h_2(y), y) - Q(h_1(y), y) \Big) \mathrm{d}y
$$

Decomposing $\oint_C Q\,\mathrm{d}y$ along boundary segments yields precisely $\int_c^d Q(h_2(y), y)\,\mathrm{d}y - \int_c^d Q(h_1(y), y)\,\mathrm{d}y$.

Adding Parts 1 and 2 yields Green's Theorem:

$$
\boxed{\oint_C (P\,\mathrm{d}x + Q\,\mathrm{d}y) = \iint_D \left( \frac{\partial Q}{\partial x} - \frac{\partial P}{\partial y} \right) \mathrm{d}A} \quad \blacksquare
$$

---

### Proof 2b: Stokes' Theorem for a Graph Surface

> **Theorem (restricted case):** Let $S$ be given as a graph $z = g(x,y)$ over a Green-admissible region $D \subset \mathbb{R}^2$, oriented by the upward normal, with boundary curve $\partial S$ projecting to $\partial D$ under compatible orientation. Let $\mathbf{F} = (P, Q, R)$ be $C^1$ on an open set containing $S$. Then $\oint_{\partial S} \mathbf{F}\cdot\mathrm{d}\mathbf{r} = \iint_S (\nabla\times\mathbf{F})\cdot\mathbf{n}\,\mathrm{d}S$.

*Proof.* Parameterize $S$ by $\mathbf{r}(x,y) = (x, y, g(x,y))$, $(x,y) \in D$, so the normal element is $\mathbf{N} = \mathbf{r}_x \times \mathbf{r}_y = (-g_x, -g_y, 1)$. Along $\partial S$ the position is $(x,y,g(x,y))$ with $(x,y) \in \partial D$, so $\mathrm{d}z = g_x\,\mathrm{d}x + g_y\,\mathrm{d}y$ and:

$$
\oint_{\partial S} \mathbf{F}\cdot\mathrm{d}\mathbf{r} = \oint_{\partial S} P\,\mathrm{d}x + Q\,\mathrm{d}y + R\,\mathrm{d}z = \oint_{\partial D} \big(P + R g_x\big)\,\mathrm{d}x + \big(Q + R g_y\big)\,\mathrm{d}y
$$

where $P, Q, R$ on the right are evaluated at $(x, y, g(x,y))$. Applying Green's Theorem (Proof 2) to $\tilde{p} = P + Rg_x$, $\tilde{q} = Q + Rg_y$ on $D$:

$$
\oint_{\partial D} \tilde{p}\,\mathrm{d}x + \tilde{q}\,\mathrm{d}y = \iint_D \left( \frac{\partial \tilde q}{\partial x} - \frac{\partial \tilde p}{\partial y} \right) \mathrm{d}A
$$

By the chain rule, $\frac{\partial \tilde q}{\partial x} - \frac{\partial \tilde p}{\partial y} = (Q_x - P_y) + (Q_z - R_x)g_x - (P_z - R_y)g_y = (\nabla\times\mathbf{F})\cdot(-g_x, -g_y, 1) = (\nabla\times\mathbf{F})\cdot\mathbf{N}$ (the middle equality is a direct expansion of $\nabla \times \mathbf{F} = (R_y - Q_z, P_z - R_x, Q_x - P_y)$ dotted with $\mathbf{N}$, checked term by term). Hence:

$$
\oint_{\partial S} \mathbf{F}\cdot\mathrm{d}\mathbf{r} = \iint_D (\nabla\times\mathbf{F})\cdot\mathbf{N}\,\mathrm{d}A = \boxed{\iint_S (\nabla\times\mathbf{F})\cdot\mathbf{n}\,\mathrm{d}S} \quad \blacksquare
$$

A general piecewise-smooth oriented surface is covered by patches that are each graphs over some coordinate plane; summing the identity patch by patch, with adjacent patches' shared-edge line integrals cancelling by opposite orientation, extends the result to the general case stated in Section 2.7.

---

### Proof 3: Gauss's Divergence Theorem for a Solid Box Element
Consider an infinitesimal rectangular box $V = [x, x+\Delta x] \times [y, y+\Delta y] \times [z, z+\Delta z]$ aligned with coordinate axes. We calculate the net outward flux of $\mathbf{F} = P\mathbf{i} + Q\mathbf{j} + R\mathbf{k}$ through all 6 faces.

1. **Faces perpendicular to $x$-axis** ($x = x+\Delta x$ front, $x = x$ back):
   - Front face ($N = \mathbf{i}$): $\iint_{\text{front}} \mathbf{F} \cdot \mathbf{i}\,\mathrm{d}S \approx P(x+\Delta x, y, z) \Delta y \Delta z$
   - Back face ($N = -\mathbf{i}$): $\iint_{\text{back}} \mathbf{F} \cdot (-\mathbf{i})\,\mathrm{d}S \approx -P(x, y, z) \Delta y \Delta z$
   - Net $x$-flux: $\big( P(x+\Delta x, y, z) - P(x, y, z) \big) \Delta y \Delta z \approx \frac{\partial P}{\partial x} \Delta x \Delta y \Delta z$

2. **Faces perpendicular to $y$-axis** ($y = y+\Delta y$ right, $y = y$ left):
   - Net $y$-flux: $\big( Q(x, y+\Delta y, z) - Q(x, y, z) \big) \Delta x \Delta z \approx \frac{\partial Q}{\partial y} \Delta x \Delta y \Delta z$

3. **Faces perpendicular to $z$-axis** ($z = z+\Delta z$ top, $z = z$ bottom):
   - Net $z$-flux: $\big( R(x, y, z+\Delta z) - R(x, y, z) \big) \Delta x \Delta y \approx \frac{\partial R}{\partial z} \Delta x \Delta y \Delta z$

Summing all 6 faces yields the total outward flux through $\partial V$:

$$
\oiint_{\partial V} \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}S \approx \left( \frac{\partial P}{\partial x} + \frac{\partial Q}{\partial y} + \frac{\partial R}{\partial z} \right) \Delta x \Delta y \Delta z = (\nabla \cdot \mathbf{F}) \Delta V
$$

This box computation only motivates the flux-density interpretation of $\nabla \cdot \mathbf{F}$; it is not a proof, since axis-aligned boxes do not tile a curved region and the boundary error is never controlled. A rigorous proof follows the same reduction-to-iterated-integrals strategy as the Green's Theorem proof above.

**Rigorous proof for a Type I/II/III region.** Suppose $V$ can be described simultaneously as: (Type III, projecting onto $D_{xy}$) $V = \{(x,y,z): (x,y) \in D_{xy},\ u_1(x,y) \le z \le u_2(x,y)\}$, and analogously as Type I and Type II regions projecting onto $D_{yz}$ and $D_{xz}$. We prove the three component identities separately and add them, exactly as Proof 2 splits Green's Theorem into a $P\,\mathrm{d}x$ identity and a $Q\,\mathrm{d}y$ identity.

*Claim:* $\displaystyle \iiint_V \frac{\partial R}{\partial z}\,\mathrm{d}V = \oiint_{\partial V} R\,\mathbf{k}\cdot\mathbf{n}\,\mathrm{d}S$.

By Fubini and the FTC applied to the inner $z$-integral over the Type III description:

$$
\iiint_V \frac{\partial R}{\partial z}\,\mathrm{d}V = \iint_{D_{xy}} \left( \int_{u_1(x,y)}^{u_2(x,y)} \frac{\partial R}{\partial z}\,\mathrm{d}z \right) \mathrm{d}A = \iint_{D_{xy}} \Big( R(x,y,u_2(x,y)) - R(x,y,u_1(x,y)) \Big)\,\mathrm{d}A
$$

The boundary $\partial V$ splits into the top graph $z = u_2(x,y)$, the bottom graph $z = u_1(x,y)$, and a lateral surface on which the outward normal is horizontal ($\mathbf{k} \cdot \mathbf{n} = 0$, so it contributes nothing to $R\,\mathbf{k}\cdot\mathbf{n}$). For a graph $z = u(x,y)$ parameterized by $(x,y)$, the normal element is $\mathbf{N} = (-u_x, -u_y, 1)$, so $\mathbf{k}\cdot\mathbf{N} = 1$ pointwise; on the top surface this $\mathbf{N}$ already points upward (outward), while on the bottom surface the outward normal is $-\mathbf{N}$. Hence:

$$
\oiint_{\partial V} R\,\mathbf{k}\cdot\mathbf{n}\,\mathrm{d}S = \iint_{D_{xy}} R(x,y,u_2(x,y))\,\mathrm{d}A - \iint_{D_{xy}} R(x,y,u_1(x,y))\,\mathrm{d}A
$$

which matches the volume integral above, proving the claim. By the analogous argument using the Type I and Type II descriptions, $\iiint_V \frac{\partial P}{\partial x}\,\mathrm{d}V = \oiint_{\partial V} P\,\mathbf{i}\cdot\mathbf{n}\,\mathrm{d}S$ and $\iiint_V \frac{\partial Q}{\partial y}\,\mathrm{d}V = \oiint_{\partial V} Q\,\mathbf{j}\cdot\mathbf{n}\,\mathrm{d}S$. Summing all three:

$$
\boxed{\oiint_{\partial V} \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}S = \iiint_V (\nabla \cdot \mathbf{F})\,\mathrm{d}V} \quad \blacksquare
$$

**Extension to general regions.** A general compact region $V$ can be partitioned into finitely many Type I/II/III pieces glued along shared internal faces. On each shared face the two adjacent pieces induce opposite outward normals ($\mathbf{n}_1 = -\mathbf{n}_2$), so their flux contributions cancel when the identity is summed over all pieces, leaving only the flux through $\partial V$ — the same additivity argument used to extend Green's Theorem beyond Type I/II regions.

---

### Proof 4: Fundamental Vector Identities

#### Identity A: $\nabla \cdot (\nabla \times \mathbf{F}) = 0$ (Divergence of Curl is Zero)
Let $\mathbf{F} = P\mathbf{i} + Q\mathbf{j} + R\mathbf{k}$ be a $C^2$ vector field.

$$
\nabla \times \mathbf{F} = \left( \frac{\partial R}{\partial y} - \frac{\partial Q}{\partial z} \right)\mathbf{i} + \left( \frac{\partial P}{\partial z} - \frac{\partial R}{\partial x} \right)\mathbf{j} + \left( \frac{\partial Q}{\partial x} - \frac{\partial P}{\partial y} \right)\mathbf{k}
$$

Taking the divergence:

$$
\nabla \cdot (\nabla \times \mathbf{F}) = \frac{\partial}{\partial x}\left( \frac{\partial R}{\partial y} - \frac{\partial Q}{\partial z} \right) + \frac{\partial}{\partial y}\left( \frac{\partial P}{\partial z} - \frac{\partial R}{\partial x} \right) + \frac{\partial}{\partial z}\left( \frac{\partial Q}{\partial x} - \frac{\partial P}{\partial y} \right)
$$

Expanding partial derivatives:

$$
\nabla \cdot (\nabla \times \mathbf{F}) = \frac{\partial^2 R}{\partial x \partial y} - \frac{\partial^2 Q}{\partial x \partial z} + \frac{\partial^2 P}{\partial y \partial z} - \frac{\partial^2 R}{\partial y \partial x} + \frac{\partial^2 Q}{\partial z \partial x} - \frac{\partial^2 P}{\partial z \partial y}
$$

By Clairaut's Theorem ($C^2$ smooth continuity allows swapping mixed partials):

$$
\frac{\partial^2 R}{\partial x \partial y} = \frac{\partial^2 R}{\partial y \partial x}, \quad \frac{\partial^2 Q}{\partial x \partial z} = \frac{\partial^2 Q}{\partial z \partial x}, \quad \frac{\partial^2 P}{\partial y \partial z} = \frac{\partial^2 P}{\partial z \partial y}
$$

Every term cancels pairwise. Thus:

$$
\boxed{\nabla \cdot (\nabla \times \mathbf{F}) = 0} \quad \blacksquare
$$

#### Identity B: $\nabla \times (\nabla f) = \mathbf{0}$ (Curl of Gradient is Zero)
Let $f$ be a $C^2$ scalar field.

$$
\nabla \times (\nabla f) = \begin{vmatrix} \mathbf{i} & \mathbf{j} & \mathbf{k} \\ \frac{\partial}{\partial x} & \frac{\partial}{\partial y} & \frac{\partial}{\partial z} \\ \frac{\partial f}{\partial x} & \frac{\partial f}{\partial y} & \frac{\partial f}{\partial z} \end{vmatrix} = \left( \frac{\partial^2 f}{\partial y \partial z} - \frac{\partial^2 f}{\partial z \partial y} \right)\mathbf{i} + \left( \frac{\partial^2 f}{\partial z \partial x} - \frac{\partial^2 f}{\partial x \partial z} \right)\mathbf{j} + \left( \frac{\partial^2 f}{\partial x \partial y} - \frac{\partial^2 f}{\partial y \partial x} \right)\mathbf{k} = \mathbf{0}
$$

$$
\boxed{\nabla \times (\nabla f) = \mathbf{0}} \quad \blacksquare
$$

---

### Proof 5: The Helmholtz-Hodge Decomposition Theorem

> **Theorem**: Any sufficiently smooth, rapidly decaying vector field $\mathbf{F}: \mathbb{R}^3 \to \mathbb{R}^3$ can be uniquely decomposed into an irrotational (curl-free) component and a solenoidal (divergence-free) component:
>
> $$
> \mathbf{F} = -\nabla \phi + \nabla \times \mathbf{A}
> $$
>
> where $\phi$ is a scalar potential and $\mathbf{A}$ is a vector potential satisfying $\nabla \cdot \mathbf{A} = 0$.

*Derivation*:
1. Take the divergence of both sides:

$$
\nabla \cdot \mathbf{F} = -\nabla \cdot (\nabla \phi) + \nabla \cdot (\nabla \times \mathbf{A}) = -\nabla^2 \phi + 0
$$

   This yields Poisson's equation for the scalar potential $\phi$:

$$
\nabla^2 \phi = -\nabla \cdot \mathbf{F}
$$

   Using the Newtonian potential Green's function for $-\nabla^2$, $\phi$ is solved explicitly by:

$$
\phi(\mathbf{r}) = \frac{1}{4\pi} \iiint_{\mathbb{R}^3} \frac{\nabla' \cdot \mathbf{F}(\mathbf{r}')}{\|\mathbf{r} - \mathbf{r}'\|}\,\mathrm{d}^3\mathbf{r}'
$$

2. Take the curl of both sides:

$$
\nabla \times \mathbf{F} = -\nabla \times (\nabla \phi) + \nabla \times (\nabla \times \mathbf{A}) = \mathbf{0} + \nabla(\nabla \cdot \mathbf{A}) - \nabla^2 \mathbf{A}
$$

   Enforcing the Coulomb gauge choice $\nabla \cdot \mathbf{A} = 0$, we get vector Poisson equations for each component of $\mathbf{A}$:

$$
\nabla^2 \mathbf{A} = -\nabla \times \mathbf{F}
$$

   Solving via Green's function:

$$
\mathbf{A}(\mathbf{r}) = \frac{1}{4\pi} \iiint_{\mathbb{R}^3} \frac{\nabla' \times \mathbf{F}(\mathbf{r}')}{\|\mathbf{r} - \mathbf{r}'\|}\,\mathrm{d}^3\mathbf{r}'
$$

$$
\boxed{\mathbf{F} = -\nabla \phi + \nabla \times \mathbf{A}} \quad \blacksquare
$$

The derivation above only shows what $\phi$ and $\mathbf{A}$ **must** be if a decomposition exists (necessity); it does not yet show that these particular $\phi, \mathbf{A}$ actually reproduce $\mathbf{F}$ (sufficiency), nor that the decomposition is unique.

*Sufficiency.* Substitute the constructed potentials back: $-\nabla\phi + \nabla\times\mathbf{A} = -\nabla\phi - \nabla^2\mathbf{A} + \nabla(\nabla\cdot\mathbf{A})$ using the vector identity $\nabla\times(\nabla\times\mathbf{A}) = \nabla(\nabla\cdot\mathbf{A}) - \nabla^2\mathbf{A}$ (Proof 4 extends componentwise to this identity). With the Coulomb gauge $\nabla\cdot\mathbf{A}=0$ and $\nabla^2\mathbf{A} = -\nabla\times\mathbf{F}$, this is $-\nabla\phi + \nabla\times\mathbf{F}$... which is not directly $\mathbf{F}$ by inspection; the reproduction instead relies on the defining property of the Newtonian Green's function $G(\mathbf{r}) = \frac{1}{4\pi\|\mathbf{r}\|}$, namely $\nabla^2 G = -\delta(\mathbf{r})$, so that $\phi = G * (\nabla\cdot\mathbf{F})$ satisfies $\nabla^2\phi = -\nabla\cdot\mathbf{F}$ *exactly* (not just formally), and likewise $\mathbf{A} = G*(\nabla\times\mathbf{F})$ satisfies $\nabla^2\mathbf{A} = -\nabla\times\mathbf{F}$ exactly; substituting these exact identities is what makes $-\nabla\phi+\nabla\times\mathbf{A}=\mathbf{F}$ hold, given that $\mathbf{F}$ decays fast enough at infinity for the convolution and the identity $\nabla(\nabla\cdot\mathbf{F})-\nabla^2\mathbf{F} = \nabla\times(\nabla\times\mathbf{F})$ to be applied validly (Helmholtz 1858; see Griffiths, *Introduction to Electrodynamics*, App. B, for the full argument).

*Uniqueness.* Suppose $\mathbf{F} = -\nabla\phi_1 + \nabla\times\mathbf{A}_1 = -\nabla\phi_2 + \nabla\times\mathbf{A}_2$ with both decompositions decaying at infinity. Then $\nabla(\phi_1-\phi_2) = \nabla\times(\mathbf{A}_1-\mathbf{A}_2)$, so $h := \phi_1-\phi_2$ is both a gradient and (by Identity A of Proof 4) has $\nabla\cdot(\nabla h) = \nabla\cdot(\nabla\times(\mathbf{A}_1-\mathbf{A}_2)) = 0$, i.e. $h$ is harmonic on all of $\mathbb{R}^3$ and $\to 0$ at infinity by the decay hypothesis; a bounded harmonic function on $\mathbb{R}^3$ vanishing at infinity is identically zero (maximum principle), so $\phi_1=\phi_2$ and hence $\nabla\times\mathbf{A}_1=\nabla\times\mathbf{A}_2$. **The decay hypothesis is essential**: without it (e.g. $\mathbf{F}$ constant) $h$ can be any harmonic function and uniqueness fails.

---

## 4. Computational & Algorithmic Insights

### 4.1 Numerical Integration & Quadrature
For complex geometries where analytical evaluation of line and surface integrals is impossible:
- **Parametric Curve Discretization**: A curve $C$ parameterized by $\mathbf{r}(t)$ for $t \in [a, b]$ is discretized into $N$ segments with nodes $t_k = a + k\Delta t$. Line integrals are approximated via Gauss-Legendre quadrature:

$$
\int_C \mathbf{F} \cdot \mathrm{d}\mathbf{r} \approx \sum_{k=1}^N w_k \mathbf{F}(\mathbf{r}(t_k)) \cdot \mathbf{r}'(t_k) \Delta t
$$

- **Triangulated Surface Flux**: A 3D surface $S$ discretized into $M$ flat triangular facets $T_j$ with normal vectors $\mathbf{N}_j = (\mathbf{v}_{j2} - \mathbf{v}_{j1}) \times (\mathbf{v}_{j3} - \mathbf{v}_{j1})$:

$$
\iint_S \mathbf{F} \cdot \mathbf{n}\,\mathrm{d}S \approx \sum_{j=1}^M \mathbf{F}(\mathbf{c}_j) \cdot \mathbf{N}_j
$$

  where $\mathbf{c}_j$ is the centroid of triangle $T_j$.

### 4.2 Finite Volume Method (FVM)
In numerical fluid dynamics and heat transfer, the **Finite Volume Method** applies Gauss's Divergence Theorem directly to control volumes $V_i$:

$$
\frac{\partial}{\partial t} \iiint_{V_i} \rho\,\mathrm{d}V + \oiint_{\partial V_i} (\rho \mathbf{v}) \cdot \mathbf{n}\,\mathrm{d}S = 0
$$

By replacing volume derivatives with discrete face fluxes across adjacent volume cell interfaces, FVM guarantees **exact mass, momentum, and energy conservation** at the machine-precision level.

### 4.3 Discrete Exterior Calculus (DEC)
In computational graphics and numerical analysis, fields are represented on discrete simplicial complexes (meshes):
- Scalar values live on 0-simplices (vertices).
- Circulation line integrals live on 1-simplices (edges).
- Flux integrals live on 2-simplices (faces).
This exact discretization preserves identity laws like $\operatorname{div}(\operatorname{curl}) = 0$ and $\operatorname{curl}(\operatorname{grad}) = 0$ without discretization error!

---

## 5. Real-World Physics & AI/ML Applications

### 5.1 Electromagnetism: Maxwell's Equations
The four fundamental equations of classical electrodynamics demonstrate the power of vector field theorems:

| Differential Form | Integral Form | Theorem Used | Physical Interpretation |
|---|---|---|---|
| $\nabla \cdot \mathbf{E} = \frac{\rho}{\varepsilon_0}$ | $\oiint_{\partial V} \mathbf{E} \cdot \mathbf{n}\,\mathrm{d}S = \frac{Q_{\text{enclosed}}}{\varepsilon_0}$ | Gauss's Divergence | Electric charges act as sources/sinks of electric field flux. |
| $\nabla \cdot \mathbf{B} = 0$ | $\oiint_{\partial V} \mathbf{B} \cdot \mathbf{n}\,\mathrm{d}S = 0$ | Gauss's Divergence | Magnetic monopoles do not exist; magnetic flux lines form closed loops. |
| $\nabla \times \mathbf{E} = -\frac{\partial \mathbf{B}}{\partial t}$ | $\oint_{\partial S} \mathbf{E} \cdot \mathrm{d}\mathbf{r} = -\frac{\mathrm{d}}{\mathrm{d}t} \iint_S \mathbf{B} \cdot \mathbf{n}\,\mathrm{d}S$ | Stokes' Theorem | Changing magnetic flux induces a circulating electric field (Faraday's Law). |
| $\nabla \times \mathbf{B} = \mu_0 \mathbf{J} + \mu_0 \varepsilon_0 \frac{\partial \mathbf{E}}{\partial t}$ | $\oint_{\partial S} \mathbf{B} \cdot \mathrm{d}\mathbf{r} = \mu_0 I_{\text{enc}} + \mu_0 \varepsilon_0 \frac{\mathrm{d}}{\mathrm{d}t} \iint_S \mathbf{E} \cdot \mathbf{n}\,\mathrm{d}S$ | Stokes' Theorem | Electric currents and changing electric fields induce magnetic circulation (Ampere-Maxwell Law). |

---

### 5.2 Fluid Dynamics & The Continuity Equation
Consider a fluid with spatial density $\rho(\mathbf{x}, t)$ and velocity field $\mathbf{v}(\mathbf{x}, t)$. The rate of mass loss inside a fixed volume $V$ equals the net mass outward flux through boundary $\partial V$:

$$
-\frac{\mathrm{d}}{\mathrm{d}t} \iiint_V \rho\,\mathrm{d}V = \oiint_{\partial V} (\rho \mathbf{v}) \cdot \mathbf{n}\,\mathrm{d}S
$$

Applying Gauss's Divergence Theorem to the surface integral:

$$
\iiint_V \left( \frac{\partial \rho}{\partial t} + \nabla \cdot (\rho \mathbf{v}) \right) \mathrm{d}V = 0
$$

Since this holds for any arbitrary volume $V$, the integrand must vanish identically, yielding the **Continuity Equation**:

$$
\frac{\partial \rho}{\partial t} + \nabla \cdot (\rho \mathbf{v}) = 0
$$

For incompressible fluids ($\rho = \text{constant}$), this reduces to the solenoid condition $\nabla \cdot \mathbf{v} = 0$.

---

### 5.3 AI & Machine Learning Connections

#### 1. Non-Conservative Optimizer Fields in Multi-Agent Learning
In single-agent Machine Learning, updates follow a conservative gradient field $\mathbf{V}(\theta) = -\nabla_\theta L(\theta)$. Since $\nabla \times \mathbf{V} = \mathbf{0}$, updates are guaranteed to be path-independent and energy-decreasing.

In multi-agent learning (e.g., GANs, zero-sum games with parameters $(\theta_1, \theta_2)$):

$$
\mathbf{V}(\theta_1, \theta_2) = \begin{pmatrix} -\nabla_{\theta_1} L(\theta_1, \theta_2) \\ +\nabla_{\theta_2} L(\theta_1, \theta_2) \end{pmatrix}
$$

The curl of this update vector field is non-zero ($\nabla \times \mathbf{V} \neq \mathbf{0}$). By Helmholtz decomposition, $\mathbf{V} = -\nabla \phi + \mathbf{V}_{\text{rot}}$. The rotational solenoidal component $\mathbf{V}_{\text{rot}}$ causes **orbiting, limit cycles, and non-convergence** during gradient descent-ascent! Modern game-theoretic optimization uses symplectic gradient adjustment to neutralize curl.

#### 2. Continuous Normalizing Flows (CNFs) & Neural ODEs
Continuous Normalizing Flows transform a simple probability distribution $p_0(\mathbf{z})$ into a complex data distribution $p_T(\mathbf{z})$ using a neural network velocity field $\mathbf{f}(\mathbf{z}, t) = \frac{\mathrm{d}\mathbf{z}}{\mathrm{d}t}$.

The log-density evolution along the trajectory is governed precisely by the **Divergence of the Neural Field**:

$$
\frac{\mathrm{d}}{\mathrm{d}t} \log p_t(\mathbf{z}(t)) = -\nabla \cdot \mathbf{f}(\mathbf{z}(t), t)
$$

Using the Instantaneous Change of Variables Formula, sampling and density evaluation are computed via vector field divergence!

---

## 5.4 Numerical Verification of the Field Theorems

The next cells check the three field theorems and the Helmholtz decomposition numerically on concrete fields, and the sign correction to the GAN-dynamics Hamiltonian derived in Exercise Problem 40.

In [1]:
import numpy as np
import matplotlib.pyplot as plt

plt.rcParams.update({
    "figure.figsize": (7.0, 4.0),
    "figure.dpi": 110,
    "axes.grid": True,
    "grid.alpha": 0.3,
    "axes.spines.top": False,
    "axes.spines.right": False,
    "font.size": 11,
})
rng = np.random.default_rng(0)
np.set_printoptions(precision=4, suppress=True)

Verify **Gauss's Divergence Theorem** on the field $\mathbf{F} = (x^3, y^3, z^3)$ over the unit ball: compare $\oiint_{\partial V}\mathbf{F}\cdot\mathbf{n}\,\mathrm{d}S$ (Monte Carlo surface sampling) against $\iiint_V \nabla\cdot\mathbf{F}\,\mathrm{d}V = \iiint_V 3(x^2+y^2+z^2)\,\mathrm{d}V$, computed exactly by switching to spherical coordinates.

In [2]:
def F(x, y, z):
    return np.array([x**3, y**3, z**3])

# Volume integral of div F = 3(x^2+y^2+z^2) over the unit ball, exact via spherical shells:
# integral_0^1 3 r^2 * 4 pi r^2 dr = 12 pi / 5
vol_integral_exact = 12 * np.pi / 5

# Surface flux via Monte Carlo over the unit sphere (n = r on the unit sphere)
n_samples = 400_000
v = rng.normal(size=(n_samples, 3))
n_hat = v / np.linalg.norm(v, axis=1, keepdims=True)
flux_vals = np.einsum("ij,ij->i", F(*n_hat.T).T, n_hat)
surface_flux_mc = flux_vals.mean() * (4 * np.pi)  # area of unit sphere = 4*pi

print(f"Volume integral (exact)      : {vol_integral_exact:.6f}")
print(f"Surface flux (Monte Carlo)   : {surface_flux_mc:.6f}")
print(f"Relative error               : {abs(surface_flux_mc - vol_integral_exact) / vol_integral_exact:.4f}")

Volume integral (exact)      : 7.539822
Surface flux (Monte Carlo)   : 7.541965
Relative error               : 0.0003


The Monte Carlo flux estimate agrees with the exact volume integral $12\pi/5 \approx 7.5398$ to within a few percent (sampling noise), confirming Gauss's Divergence Theorem for a field that is genuinely 3D (not the box-heuristic case).

In [3]:
# Verify Stokes' Theorem on the graph surface S: z = g(x,y) = x^2 + y^2 over the unit disk D,
# with F = (-y, x, x*y). Compare the line integral around boundary(S) to the surface flux of curl F.
def Fvec(x, y, z):
    return np.array([-y, x, x * y])

def g(x, y):
    return x**2 + y**2

# Boundary curve: unit circle lifted onto S, r(t) = (cos t, sin t, 1) for t in [0, 2pi]
t = np.linspace(0, 2 * np.pi, 200_000)
rx, ry, rz = np.cos(t), np.sin(t), np.ones_like(t)
drx, dry, drz = -np.sin(t), np.cos(t), np.zeros_like(t)
Fx, Fy, Fz = Fvec(rx, ry, rz)
line_integral = np.trapezoid(Fx * drx + Fy * dry + Fz * drz, t)

# curl F = (x, -y, 2) for this F; surface integral over the graph z=g(x,y), N = (-g_x,-g_y,1) = (-2x,-2y,1)
n_grid = 800
u = np.linspace(-1, 1, n_grid)
X, Y = np.meshgrid(u, u)
mask = X**2 + Y**2 <= 1
curl_dot_N = X * (-2 * X) + (-Y) * (-2 * Y) + 2 * 1
cell_area = (u[1] - u[0]) ** 2
surface_integral = curl_dot_N[mask].sum() * cell_area

print(f"Line integral around boundary(S) : {line_integral:.6f}")
print(f"Surface integral of curl F . n   : {surface_integral:.6f}")
print(f"Absolute difference              : {abs(line_integral - surface_integral):.6f}")

Line integral around boundary(S) : 6.283185
Surface integral of curl F . n   : 6.281995
Absolute difference              : 0.001190


The two computations agree to grid/quadrature resolution, confirming Stokes' Theorem for the graph-surface proof above ($\iint_S \nabla\times\mathbf{F}\cdot\mathbf{n}\,\mathrm{d}S = \oint_{\partial S}\mathbf{F}\cdot\mathrm{d}\mathbf{r}$).

In [4]:
# Verify the corrected Exercise Problem 40 identity: J @ grad(H) = K @ x for H = (b-c)/4 (x^2+y^2),
# J = [[0,1],[-1,0]], K the skew-symmetric part of A = [[a,b],[c,d]].
a, b, c, d = 2.0, -1.5, 3.0, 0.5
A = np.array([[a, b], [c, d]])
K = np.array([[0, (b - c) / 2], [(c - b) / 2, 0]])
J = np.array([[0, 1], [-1, 0]])

x_test = rng.normal(size=2)
grad_H = (b - c) / 2 * x_test  # gradient of H = (b-c)/4 (x^2+y^2)
lhs = J @ grad_H
rhs = K @ x_test
print(f"J @ grad(H)        : {lhs}")
print(f"K @ x               : {rhs}")
print(f"Max abs difference  : {np.max(np.abs(lhs - rhs)):.2e}")

J @ grad(H)        : [-1.7388 -1.43  ]
K @ x               : [-1.7388 -1.43  ]
Max abs difference  : 0.00e+00


With the corrected Hamiltonian $H = \frac{b-c}{4}(x^2+y^2)$ (previously stated with the opposite sign), $\mathbf{J}\nabla H$ matches $K\mathbf{x}$ to machine precision, confirming the fix to Exercise Problem 40.

---

## 6. Canonical Literature Mapping & References

- **Marsden & Tromba, *Vector Calculus***: Chapters 4 (Acceleration & Vector Fields), 6 (Line & Surface Integrals), 7 (Green's, Stokes', and Gauss's Theorems).
- **Apostol, *Calculus Vol. II***: Chapter 10 (Line Integrals), Chapter 11 (Surface Integrals & Field Theorems).
- **Spivak, *Calculus on Manifolds***: Chapter 4 (Integration & Chains), Chapter 5 (Stokes' Theorem on Manifolds).
- **Stewart, *Multivariable Calculus***: Chapter 16 (Vector Calculus).
- **Demidovich, *Problems in Mathematical Analysis***: Section VIII (Line and Surface Integrals, Green/Stokes/Gauss Formulae).